src/processor/operator/aggregate/aggregate_hash_table.cpp

```diff
@@ -230,57 +239,75 @@ void AggregateHashTable::initializeFTEntries(const std::vector<ValueVector*>& fl
     }
     for (auto i = 0u; i < numFTEntriesToInitialize; i++) {
         auto entryIdx = entryIdxesToInitialize[i];
-        auto entry = hashSlotsToUpdateAggState[entryIdx]->entry;
-        fillEntryWithInitialNullAggregateState(entry);
+        auto& slot = *hashSlotsToUpdateAggState[entryIdx];
+        fillEntryWithInitialNullAggregateState(*factorizedTable, slot.entry);
+        // Fill the hashValue in the ftEntry.
+        factorizedTable->updateFlatCellNoNull(slot.entry, hashColIdxInFT,
+            hashVector->getData() + hashVector->getNumBytesPerValue() * entryIdx);
+    }
+}
+
+void AggregateHashTable::initializeFTEntries(const FactorizedTable& sourceTable,
+    uint64_t sourceStartOffset, uint64_t numFTEntriesToInitialize) {
+    // auto colIdx = 0u;
+    for (size_t i = 0; i < numFTEntriesToInitialize; i++) {
+        auto idx = entryIdxesToInitialize[i];
+        auto& slot = *hashSlotsToUpdateAggState[idx];
+        auto sourcePos = sourceStartOffset + idx;
+        memcpy(slot.entry, sourceTable.getTuple(sourcePos),
+            getTableSchema()->getNumBytesPerTuple());
+        // TODO: Ideally we should actually copy the overflow so that the original overflow data can
```

Comment:
What's preventing this TODO? Are you going to address it separately? or not really looking forward to address it any soon?

Reply:
It might have a significant impact on performance, so I wasn't rushing to complete it, but it probably should be addressed.

Reply:
I think this would only work if we can get the overflow to support concurrent appends. At this point in the code that wouldn't matter, but we still wouldn't be able to free the overflow unless we also copy the overflow when partitioning. 
Concurrent appends should be possible, but I was seeing a reasonably large difference in runtime, so maybe best to just leave it for now (as big as 0.5s->0.9s on one query with long strings).